### PR TITLE
More analyzer types checks

### PR DESCRIPTION
**Pull request by jgruica**

### CODE REVIEW

File: pkg/analyze/download.go [hunk 1]
```diff
@@ -188,14 +188,15 @@ func parseAnalyzers(spec string) ([]*troubleshootv1beta2.Analyze, error) {
...
+	if gvk.Group == "troubleshoot.sh" && gvk.Kind == "SupportBundle" {
 		supportBundle := obj.(*troubleshootv1beta2.SupportBundle)
 		analyzer := supportBundleToAnalyzer(supportBundle)
 		return analyzer.Spec.Analyzers, nil
-	} else {
+	} else if gvk.Group == "troubleshoot.sh" && gvk.Kind == "Analyzer" {
 		analyzer := obj.(*troubleshootv1beta2.Analyzer)
 		return analyzer.Spec.Analyzers, nil
 	}
+	return nil, nil
```
Comment (divolgin) [inline pkg/analyze/download.go]: I don't think the callers handle nil values

File: pkg/analyze/download.go [hunk 2]
```diff
@@ -188,14 +188,15 @@ func parseAnalyzers(spec string) ([]*troubleshootv1beta2.Analyze, error) {
 	}
 
 	// SupportBundle overwrites Analyzer if defined
-	if gvk.String() == "troubleshoot.sh/v1beta2, Kind=SupportBundle" {
+	if gvk.Group == "troubleshoot.sh" && gvk.Kind == "SupportBundle" {
 		supportBundle := obj.(*troubleshootv1beta2.SupportBundle)
 		analyzer := supportBundleToAnalyzer(supportBundle)
```
Comment (divolgin) [inline pkg/analyze/download.go]: ~analyzer := supportBundleToAnalyzer(supportBundle)~
return supportBundle.Spec.Analyzer, nil

File: pkg/analyze/download.go [hunk 3]
```diff
@@ -188,14 +188,15 @@ func parseAnalyzers(spec string) ([]*troubleshootv1beta2.Analyze, error) {
 	}
 
 	// SupportBundle overwrites Analyzer if defined
-	if gvk.String() == "troubleshoot.sh/v1beta2, Kind=SupportBundle" {
+	if gvk.Group == "troubleshoot.sh" && gvk.Kind == "SupportBundle" {
```
Comment (divolgin) [inline pkg/analyze/download.go]: this should still check `v1beta2` since cast uses v1beta2

File: pkg/analyze/download.go [hunk 4]
```diff
@@ -188,29 +186,15 @@ func parseAnalyzers(spec string) ([]*troubleshootv1beta2.Analyze, error) {
...
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("%s-analyzer", sb.Name),
-		},
-		Spec: troubleshootv1beta2.AnalyzerSpec{
-			Analyzers: sb.Spec.Analyzers,
-		},
-	}
+	
+	return nil, errors.Wrapf(err, "invalid analyzer %q", gvk)
```
Comment (emosbaugh) [inline pkg/analyze/download.go]: maybe
```
errors.Wrapf(err, "invalid gvk %q", gvk)
```

**Review by emosbaugh — APPROVED** [review]

File: pkg/analyze/download.go (hunk 4 above)
Comment (divolgin) [reply pkg/analyze/download.go]: `errors.Errorf` is probably what we want here :)

Comment (jgruica) [reply pkg/analyze/download.go]: @divolgin updated

File: pkg/analyze/download.go [hunk 5]
```diff
@@ -188,29 +186,15 @@ func parseAnalyzers(spec string) ([]*troubleshootv1beta2.Analyze, error) {
...
-			Kind:       "Analyzer",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("%s-analyzer", sb.Name),
-		},
-		Spec: troubleshootv1beta2.AnalyzerSpec{
-			Analyzers: sb.Spec.Analyzers,
-		},
-	}
+	return nil, errors.Errorf("Invalid gvk %q", gvk)
```
Comment (divolgin) [inline pkg/analyze/download.go]: lower case `I`

**Review by divolgin — APPROVED** [review]

**Pull request merged**